static/js/components/tiles/disaster_event_map_tile.tsx

```diff
@@ -277,219 +279,333 @@ export const DisasterEventMapTile = memo(function DisasterEventMapTile(
-          eventTypeGeoJsonData[type] = geoJsonData;
-        });
-        setEventTypeGeoJson(eventTypeGeoJsonData);
-      })
-      .catch(() => {
-        setEventTypeGeoJson({});
-      });
+    // if at least one event type feature was read from an event point, then
+    // all event points should contain geojson feature information IF its
+    // available for the event point.
+    const eventDcidsToFetch = _.isEmpty(eventTypeFeatures[eventType])
+      ? eventDcids
+      : [];
+    geoJsonPromises.push(
+      fetchNodeGeoJson(
+        eventDcidsToFetch,
+        props.eventTypeSpec[eventType][geoJsonPropKey],
+        apiRoot
+      )
+    );
   }
+  return Promise.all(geoJsonPromises)
+    .then((geoJsons) => {
+      const eventTypeGeoJsonData = {};
+      eventTypeKeys.forEach((type, i) => {
+        if (!geoJsons[i]) {
+          return;
+        }
+        const geoJsonData = geoJsons[i];
+        geoJsonData.features.push(...eventTypeFeatures[type]);
+        eventTypeGeoJsonData[type] = geoJsonData;
+      });
+      return eventTypeGeoJsonData;
+    })
+    .catch(() => {
+      return {};
+    });
+}
 
-  /**
-   * Draws the disaster event map
-   */
-  function draw(
-    parentPlaces: NamedPlace[],
-    geoJsonData: GeoJsonData,
-    disasterEventData: Record<string, DisasterEventPointData>,
-    polygonGeoJson: GeoJsonData,
-    pathGeoJson: GeoJsonData,
-    zoomParams: MapZoomParams
-  ): void {
-    const width = svgContainerRef.current.offsetWidth;
-    const height = svgHeight;
-    const isUsaPlace = isChildPlaceOf(
-      props.place.dcid,
-      USA_PLACE_DCID,
-      parentPlaces
-    );
-    const isPlaceBaseMap = shouldUsePlaceGeoJson();
-    const projection = getProjection(
-      isUsaPlace,
-      props.place.dcid,
-      width,
-      height,
-      geoJsonData,
-      props.place.dcid
-    );
-    drawD3Map(
-      svgContainerRef.current,
-      geoJsonData,
-      height,
-      width,
-      {} /* dataValues: no data values to show on the base map */,
-      null /* colorScale: no color scale since no data shown on the base map */,
-      (geoFeature: GeoJsonFeatureProperties) =>
-        redirectAction(geoFeature.geoDcid) /* redirectAction */,
-      (place: NamedPlace) => place.name || place.dcid /* getTooltipHtml */,
-      (placeDcid: string) =>
-        placeDcid !==
-        props.place
-          .dcid /* canClickRegion: don't allow clicking region that will redirect to current page */,
-      true /* shouldShowBoundaryLines */,
-      projection,
-      isPlaceBaseMap
-        ? ""
-        : props.place
-            .dcid /** zoomDcid: don't want special zoom handling of the selected place if usiing place base map */,
-      zoomParams
+/**
+ * Fetches the chart data for a disaster event map
+ * @param props props for the disaster event map tile
+ * @param fetchData function to use to fetch data with a given cache key and data promise
+ */
+export async function fetchChartData(
```

Comment:
`async` only necessary if `await` is called in the body.

Reply:
done